### PR TITLE
Extract renderer object

### PR DESCRIPTION
**Pull request by georgebrock**
- Adds better rendering tests.
- Extracts the rendering code from `Base` into `Renderer`.
- Replaces the `Demo` object with a `DemoRenderer`.
- Fixes command line rendering for search.

### CODE REVIEW

File: js/demo_renderer.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            if (li.length === 0 || li.hasClass('complete')) {
+                li = $('<li></li>').prependTo(this.commandList);
+            }
+            li.html(op.description(vim))
+              .toggleClass('complete', op.complete() || op.cancelled);
+        }
+    };
+
+    delegateMethod('readTextContainer');
+    delegateMethod('bindKeyListener');
```
Comment (mike-burns) [inline js/demo_renderer.js:33]: This seems worth blogging about.

File: js/renderer.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                                                $(this).focus();
+                                            });
+
+        // Use keyup for special characters like escape
+        $(window).keyup(function (e) {
+            var code = e.charCode || e.keyCode;
+            if (
+                code === Vimulator.Utils.Keys.BACKSPACE.charCodeAt(0) ||
+                code === Vimulator.Utils.Keys.ESC.charCodeAt(0) ||
+                code === Vimulator.Utils.Keys.RETURN.charCodeAt(0)
```
Comment (mike-burns) [inline js/renderer.js]: How about pulling this out into another method.

Comment (georgebrock) [inline js/renderer.js]: Good call, @mike-burns. Done c1bc078.

**Comment by mike-burns** [comment]
Looks great to merge.

**Comment by georgebrock** [comment]
Thanks Mike, merged.

**Pull request merged**